src/ui/settings.tsx

```diff
@@ -0,0 +1,112 @@
+import * as IRF from 'internet-roadtrip-framework';
+import { render } from 'solid-js/web';
+
+import styles, { stylesheet as moduleCss } from '../style.module.css';
+import { getVehicle, setVehicle } from '../pano/vehicle';
+import { getNoiseActive, setNoiseActive, getNoiseVolume, setNoiseVolume } from '../noise/noise';
+import { getWheelRotation, setWheelRotation } from '../pano/wheel';
+import { getOptionsRotation, setOptionsRotation } from '../pano/options';
+import { getMaxZoom, setMaxZoom } from '../pano/scene';
+
+if (IRF.isInternetRoadtrip) {
+	const { container: settings } = IRF.ui.panel.createTabFor(
+		{ ...GM.info, script: { name: 'LOtW' } },
```

Comment:
Is there a need to override `script.name` here? I'd think its for the shorter tab name, but that's set right below.

Reply:
This is for the footer, not the tab name.

The script name is `Internet Roadtrip - Look Out The Window`, and we can't change this in the metadata block because everyone who has installed v1 has it installed under that name, and the way userscript managers track scripts is via a combination of name and namespace.

`Internet Roadtrip - Look Out The Window` is too long for all the contributor names to actually fit in the footer aswell.

So with the existing name, the footer is too long, and we can't change the name in the metadata block, so it's changed here to remove `Internet Roadtrip - `.

Reply:
Actually, LOtW v1 is called "Internet Roadtrip - Look Out the Window v1" (which means in the IRF panel tab's footer it shows up as "Internet Roadtrip - Look Out the Window v1 v1.y.z by netux" 😆).

Except the IRF panel footer does not actually show that name, because I've been programmatically removing the "Internet Roadtrip - " part.

---

I just tested it, and "Look Out the Window" fits well, even in smaller screen sizes:

![](https://github.com/user-attachments/assets/819b2366-6e73-4952-9df1-03fc0fc2a5a5)

I'm fine with just using my username in the contributors list, so that can be reduced even further!

Reply:
So we keep this GM code change? Would you prefer to switch to using your username in the `package.json` or programatically swap it out in the `settings.tsx` so you can keep it in the `package.json`?

Reply:
This is what I've been doing in all my userscripts:

```suggestion
		{ ...GM.info, script: { ... GM.info.script, name: GM.info.script.name.replace('Internet Roadtrip - ', '') } },
```

---

Yeah, I'll simplify my name in the package.json